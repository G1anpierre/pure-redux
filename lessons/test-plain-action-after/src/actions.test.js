import {
  selectHouse,
  addPoints,
  resetScores
} from './actions';

test('resetScores', () => {
  const result = resetScores();
  expect(result.type).toEqual('RESET_SCORES');
});

test('addPoints', () => {
  const house = { id: 7 };
  const result = addPoints(house, 50);
  expect(result.type).toEqual('ADD_POINTS');
  expect(result.house).toBe(house);
  expect(result.points).toEqual(50);
  expect(result).toMatchSnapshot();
});

test('selectHouse', () => {
  const house = {};
  const result = selectHouse(house);
  expect(result.type).toEqual('SELECT_HOUSE');
  expect(result.house).toBe(house);
});
