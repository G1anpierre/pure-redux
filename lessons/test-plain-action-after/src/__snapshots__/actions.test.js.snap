// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addPoints 1`] = `
Object {
  "house": Object {
    "id": 7,
  },
  "points": 50,
  "type": "ADD_POINTS",
}
`;
